include/boost/thread/pthread/mutex.hpp

```diff
@@ -266,8 +266,7 @@ namespace boost
         // fixme: Shouldn't this functions be located on a .cpp file?
         bool do_try_lock_until(detail::internal_timespec_timepoint const &timeout)
         {
-          const timespec ts = timeout.getTs();
-          int const res=pthread_mutex_timedlock(&m,&ts);
+          int const res=pthread_mutex_timedlock(&m,&timeout.getTs());
```

Comment:
I find this weir. Passing the address of the returned value.

Reply:
That's what the original code did. getTs() passes a const reference so it's valid and saves an unnecessary copy of the variable.

Reply:
It us just I'm not used to see this kind of construct. It is just me.